### DISCUSSION TITLE
Sonoff basic with PZEM-004 configuration loosing WiFi connection after powercycle

### DISCUSSION THREAD

**Discussion by HS1979** [arendst/Tasmota#17167] [Q&A] [answered]
Hi,
I have completed power meter with use of Sonoff basic R1 and PZEM-004 using instructions from here https://tasmota.github.io/docs/PZEM-0XX/ . So actions made are only adding additional 1k resistor and connecting wires between PZEM-004 and sonoff. 
Tasmota config for sonoff is like this:
![image](https://user-images.githubusercontent.com/97183706/203513797-410a0ce0-08e8-4574-bd8a-3e407cf91c4c.png)

Everything work fine until power is lost. After powering up, Sonoff cannot connect to WiFi anymore. Sometimes in very short periods (second or two) I can see on my WiFi access point that it is connected for very shor time (1-5 seconds).

Tried with different access points, with PZEM-004 connected or not., with multiple Sonoff Basic R1 devices, always same behaviour. 

After moment when I configure Generic(18) template as on picture, Sonoff should not be powered off as it can not connect to WiFi anymore. If Sonoff is not powered off, everything works OK for days, no any issues with WiFi connection.

To me this looks like some bug in Tasmota or missconfiguration that I don't know how to resolve. Tried with 10.0 and latest 12.2 Tasmota version,

Does anybody have some hint or solution for this?

Thanks.

**Comment by Jason2866** [comment, marked as answer]
Sonoff Basic power supply is to weak to drive a PZEM-004. Power Supply issues is resulting in resetting settings.
